build/patches/add-user-scripts.v5.patch

```diff
@@ -0,0 +1,9810 @@
++
++    // remove unicode chars and set content into File
++    stripUnicode(content);
++    std::unique_ptr<UserScript::File> file(new UserScript::File());
++    file->set_content(content);
++    file->set_url(GURL(/*script_key*/ "script.js")); // name doesn't matter
++
++    // create SHA256 of file
++    char raw[crypto::kSHA256Length] = {0};
++    std::string key;
++    crypto::SHA256HashString(content, raw, crypto::kSHA256Length);
++    base::Base64Encode(base::StringPiece(raw, crypto::kSHA256Length), &key);
++    file->set_key(key);
++
++    script->js_scripts().push_back(std::move(file));
++
++    return true;
++}
++
++// static
++bool GetOrCreatePath(base::FilePath& path) {
++  base::PathService::Get(base::DIR_ANDROID_APP_DATA, &path);
++  path = path.AppendASCII("snippets");
++
++  // create snippets directory if not exists
++  if(!base::PathExists(path)) {
++    LOG(INFO) << "Path " << path << " doesn't exists. Creating";
++    base::File::Error error = base::File::FILE_OK;
++    if( !base::CreateDirectoryAndGetError(path, &error) ) {
++      LOG(INFO) <<
++               "ERROR: failed to create directory: " << path
++               << " with error code " << error;
++      return false;
++    }
++  }
++  return true;
++}
++
++// static
++void LoadUserScripts(UserScriptList* user_scripts) {
++  base::FilePath path;
++  if (GetOrCreatePath(path) == false) return;
++
++  // enumerate all files from script path
++  // we accept all files, but we check if it's a real
++  // userscript
++  base::FileEnumerator dir_enum(
++    path,
++    /*recursive=*/false, base::FileEnumerator::FILES);
++  base::FilePath full_name;
++  while (full_name = dir_enum.Next(), !full_name.empty()) {
++    std::unique_ptr<UserScript> userscript(new UserScript());
++
++    base::string16 error;
++    if (LoadUserScriptFromFile(full_name, GURL(), userscript, &error)) {
++      LOG(INFO) << "Found user script " << userscript->name() <<
++                                    "-" << userscript->version() <<
++                                    "-" << userscript->description();
++      user_scripts->push_back(std::move(userscript));
++    } else {
++      LOG(INFO) << "User script load error " << error;
++    }
++  }
++}
++
++UserScriptLoader::UserScriptLoader(BrowserContext* browser_context,
++                                   UserScriptsPrefs* prefs)
++    : loaded_scripts_(new UserScriptList()),
++      ready_(false),
++      browser_context_(browser_context),
++      prefs_(prefs) {}
++
++UserScriptLoader::~UserScriptLoader() {
++  for (auto& observer : observers_)
++     observer.OnUserScriptLoaderDestroyed(this);
++}
++
++void UserScriptLoader::OnRenderProcessHostCreated(
++    content::RenderProcessHost* process_host) {
++  if (initial_load_complete()) {
++    SendUpdate(process_host, shared_memory_);
++  }
++}
++
++void UserScriptLoader::AttemptLoad() {
++  // if (ready_ && ScriptsMayHaveChanged()) {
```

Comment:
the original code foresees that the init of the extensions is asynchronous, so that flag enables the delayed start and the possible modifications of the list of user scripts as the extensions are activated gradually.
we don't need it, it's part of the code to be eliminated

Reply:
fixed in https://github.com/bromite/bromite/pull/857/commits/2812a0f9a76a7b3052b7a408344095680e7244ff